src/python/ensembl/genes/tracking/bioproject_tracking.py

```diff
@@ -0,0 +1,250 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import requests
+import pymysql
+import json
+from pathlib import Path
+from collections import Counter
+
+with open("./bioproject_tracking_config.json", "r") as f:
+    config = json.load(f)
+
+def mysql_fetch_data(query, database, host, port, user):
+    """
+    Executes a given SQL query on a MySQL database and fetches the result.
+
+    This function establishes a connection to the MySQL database using provided connection details.
+    It then executes the given query using a cursor obtained from the connection. After executing the query,
+    it fetches all the rows of the query result and returns them. The function handles any errors that might
+    occur during the process and ensures that the database connection is closed before returning the result.
+
+    Args:
+        query (str): The SQL query to be executed.
+        database (str): The name of the database to connect to.
+        host (str): The host name or IP address of the MySQL server.
+        port (int): The port number to use for the connection.
+        user (str): The username to use for the database connection.
+
+    Returns:
+        tuple: A tuple of tuples containing the rows returned by the query execution.
+
+    Note:
+        This function does not handle database password authentication. Ensure that the provided user
+        has the necessary permissions and that the database is configured to allow password-less connections
+        from the given host.
+    """
+    try:
+        conn = pymysql.connect(
+            host=host, user=user, port=port, database=database.strip()
+        )
+
+        cursor = conn.cursor()
+        cursor.execute(query)
+        info = cursor.fetchall()
+
+    except pymysql.Error as err:
+        print(err)
+
+    cursor.close()
+    conn.close()
+    return info
+    
+def get_assembly_accessions(bioproject_id, only_haploid=False):
+    """
+    Fetches assembly accessions from a given NCBI BioProject ID.
+
+    Args:
+    bioproject_id (str): The NCBI BioProject ID.
+    only_haploid (bool): If True, fetch only haploid assemblies. Defaults to False.
+
+    Returns:
+    list: A list of assembly accessions.
+    """
+    base_url = "https://api.ncbi.nlm.nih.gov/datasets/v2alpha/genome/bioproject"
+    next_page_token = None
+    assembly_accessions = {}
+
+    while True:
+        url = f"{base_url}/{bioproject_id}/dataset_report"
+        if next_page_token:
+            url += f"?page_token={next_page_token}"
+
```

Comment:
Is this request stable? might help adding a try/except?

Reply:
Done + works